src/components/RadioButtonGroup/BaseRadioButtonGroup.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, useEffect, useState } from 'react'
+
+export const RadioButtonGroupContext = createContext<RadioButtonGroupContext>({
+  onSelectValue: () => undefined,
+  selectedValue: undefined,
+})
+
+interface RadioButtonGroupContext {
+  onSelectValue: (val: string) => void
+  selectedValue: string
+}
+
+export interface BaseRadioButtonGroupProps {
+  onCheck?: (val: string) => void
+  checked?: string
+}
+
+const BaseRadioButtonGroup: React.FC<
+  React.ComponentPropsWithoutRef<'div'> & BaseRadioButtonGroupProps
+> = ({ children, onCheck, checked, ...rest }) => {
+  const [selectedValue, setSelectedValue] = useState<string | undefined>(checked)
+
+  useEffect(() => {
+    if (selectedValue !== checked) {
+      onCheck?.(selectedValue)
+    }
+  }, [checked, selectedValue])
```

Comment:
This looks like a bug. `check` shouldn't be used for anything other than the initial value. A test that checks one option then checks another option should catch this.

Also, I wouldn't use an effect for this, instead:
```suggestion
  // pass this to the context
  const onSelectValue = useCallback((value: string) => {
    setSelectedValue(value)
    onCheck?.(value)
  }, [onCheck])
```